src/rustup/command.rs

```diff
@@ -135,24 +135,38 @@ fn telemetry_rustc<S: AsRef<OsStr>>(mut cmd: Command,
 fn run_command_for_dir_without_telemetry<S: AsRef<OsStr>>(
     mut cmd: Command, arg0: &str, args: &[S]) -> Result<()>
 {
-    cmd.args(&args);
+    #[cfg(unix)]
+    fn run(mut command: Command, arg0: &str) -> Result<()> {
+        use std::os::unix::process::CommandExt;
+        let error = command.exec();
+        Err(error).chain_err(|| rustup_utils::ErrorKind::RunningCommand {
```

Comment:
You’ll probably want to chain the error after calling run, like this:

```
run(cmd, arg0).chain_err(|| ...)
```

instead. That greatly cuts down on the amount of code needed.